src/EnergyPlus/EcoRoofManager.cc

```diff
@@ -931,11 +961,23 @@ namespace EcoRoofManager {
             // Smart schedule only irrigates when scheduled AND the soil is less than 40% saturated
             state.dataEcoRoofMgr->CurrentIrrigation = state.dataWaterData->Irrigation.ScheduledAmount; // units of m
             state.dataWaterData->Irrigation.ActualAmount = state.dataEcoRoofMgr->CurrentIrrigation;
+        } else if (state.dataWaterData->Irrigation.ModeID == DataWater::RainfallMode::Invalid) {
+            // no schedule, irrigation is just the rain amount
+            state.dataEcoRoofMgr->CurrentIrrigation = state.dataEcoRoofMgr->CurrentPrecipitation; // units of m
+            state.dataWaterData->Irrigation.ActualAmount = state.dataEcoRoofMgr->CurrentIrrigation;
```

Comment:
As I commented in the documentation section, it seems this should use the sum of `Irrigation.ScheduledAmount + CurrentPrecipitation` or add an option to use both.

Reply:
Yes, the "Use Precipitation" field is added to control whether to add the precipitation to irrigation.